### PR TITLE
add istio-gateway helpers to istio-pilot event handler rewrite

### PR DESCRIPTION
**Pull request by ca-scribner**
This is based on #234.  Recommended that this be reviewed after #234 is merged as that accounts for most of the changes.  This is part of a larger effort to refactor istio-pilot's event handling, as described [in jira](https://warthogs.atlassian.net/browse/KF-728). This merges into a feature branch and covers only part of the total effort.

Includes:
* helpers that check on istio-gateway's service
* tests for helpers
* added a config option for the istio-gateway's service name to make things easier to test and deploy

### CODE REVIEW

File: charms/istio-pilot/src/charm.py [hunk 1]
```diff
@@ -347,6 +346,31 @@ def _get_ingress_auth_data(self) -> dict:
...
+        #  from the istio-gateway.  Providing here as a config at least makes this less rigid than
+        #  assuming the name.
+        # TODO: What happens if this service does not exist?  We should check on that and then add
+        #  tests to confirm this works
+        svc = self.lightkube_client.get(
+            Service, name=self.model.config["gateway-service-name"], namespace=self.model.name
+        )
+        return svc
+
+    def _is_gateway_service_up(self):
```
Comment (beliaev-maksim) [inline charms/istio-pilot/src/charm.py:361]: maybe convert to property ?

Comment (ca-scribner) [reply charms/istio-pilot/src/charm.py:361]: I'm never sure when to use a property.  Do you have a guideline to recommend?  

My rule of thumb is to do it when there's essentially no computation behind the item.  This one has very trivial computation behind it, but lots of logic across a few helpers.  So I'm torn :D

**Comment by i-chvets** [comment]
LGTM. Let me know when approval is needed.

**Review by i-chvets — APPROVED** [review]
Approved.

File: charms/istio-pilot/src/charm.py (hunk 1 above)
Comment (beliaev-maksim) [reply charms/istio-pilot/src/charm.py:361]: I think the whole idea of properties to have computation behind :)
otherwise you use just class attribute

Comment (ca-scribner) [reply charms/istio-pilot/src/charm.py]: done!

**Review by beliaev-maksim — APPROVED** [review]

**Pull request merged**